pkg/snapshot/file/publisher_test.go

```diff
@@ -0,0 +1,132 @@
+package publisher
+
+import (
+	"context"
+	"encoding/csv"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/statediff/indexer/ipld"
+	"github.com/jackc/pgx/v4"
+
+	fixt "github.com/vulcanize/eth-pg-ipfs-state-snapshot/fixture"
+	snapt "github.com/vulcanize/eth-pg-ipfs-state-snapshot/pkg/types"
+	"github.com/vulcanize/eth-pg-ipfs-state-snapshot/test"
+)
+
+var (
+	pgConfig = test.DefaultPgConfig
+	nodeInfo = test.DefaultNodeInfo
+	// tables ordered according to fkey depedencies
+	allTables = []*snapt.Table{
+		&snapt.TableIPLDBlock,
+		&snapt.TableNodeInfo,
+		&snapt.TableHeader,
+		&snapt.TableStateNode,
+		&snapt.TableStorageNode,
+	}
+)
+
+func writeFiles(t *testing.T, dir string) *publisher {
+	pub, err := NewPublisher(dir, nodeInfo)
+	test.NoError(t, err)
+	test.NoError(t, pub.PublishHeader(&fixt.Block1_Header))
+	tx, err := pub.BeginTx()
+	test.NoError(t, err)
+
+	headerID := fixt.Block1_Header.Hash().String()
+	test.NoError(t, pub.PublishStateNode(&fixt.Block1_StateNode0, headerID, tx))
+
+	test.NoError(t, tx.Commit())
+	return pub
+}
+
+// verify that we can parse the csvs
+// TODO check actual data
```

Comment:
This is a key TODO, and something we need to do before deploying this.

I think the first approach we should take here is to run this new state of the snapshot tool at some blockheight that has state and storage trie nodes but is early enough to finish in a semi-reasonable amount of time (e.g. block 100,000). Have it write that out to csv files, then load that into Postgres, and use https://github.com/vulcanize/ipld-eth-db-validator to confirm everything is in Postgres.

This should be an automated integration test but for that we need to either port around a levelDB snapshot at/near that height or setup our LevelDB RPC server and start using that in these types of tests (ideal case).

For a first pass, and as a requirement to get this merged, you can  setup and run that test manually.